app/src/main/java/com/nextcloud/talk/diagnose/DiagnoseContentComposable.kt

```diff
@@ -59,6 +95,124 @@ fun DiagnoseContentComposable(data: State<List<DiagnoseActivity.DiagnoseElement>
+                .fillMaxWidth()
+                .padding(vertical = 8.dp),
+            fontSize = LocalDensity.current.run {
+                dimensionResource(R.dimen.headline_text_size).toPx().toSp()
+            },
+            fontWeight = FontWeight.Bold,
+            textAlign = TextAlign.Center
+        )
+    }
+}
+
+@Composable
+fun ShowOptions(onDismissDialog: () -> Unit, message: String, context: Context) {
+    Row(
+        horizontalArrangement = Arrangement.SpaceEvenly,
+        modifier = Modifier.fillMaxWidth()
+    ) {
+        TextButton(onClick = { onDismissDialog() }) {
+            Text(text = stringResource(R.string.nc_cancel))
+        }
+        Spacer(modifier = Modifier.width(8.dp))
+        TextButton(onClick = {
+            val clipboard =
+                context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+            val clip = ClipData.newPlainText("Push Message", message)
+            clipboard.setPrimaryClip(clip)
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.S_V2) {
+                Toast.makeText(context, R.string.message_copied, Toast.LENGTH_SHORT).show()
+            }
+            onDismissDialog()
+        }) {
+            Text(text = stringResource(R.string.nc_common_copy))
+        }
+    }
+}
+
+@Composable
+fun LoadingIndicator() {
+    Box(
+        modifier = Modifier.fillMaxSize(),
+        contentAlignment = Alignment.Center
+    ) {
+        CircularProgressIndicator()
+    }
+}
+
+@Composable
+fun ShowNotificationData(
+    isLoading: Boolean,
+    showDialog: Boolean,
+    context: Context,
+    message: String,
+    onDismissDialog: () -> Unit
+) {
+    if (isLoading) {
+        LoadingIndicator()
+    }
+    if (showDialog) {
+        Dialog(
+            onDismissRequest = { onDismissDialog() },
+            properties = DialogProperties(
+                dismissOnClickOutside = true,
+                usePlatformDefaultWidth = false
+            )
+        ) {
+            Surface(
+                shape = MaterialTheme.shapes.medium,
+                tonalElevation = 8.dp,
+                modifier = Modifier
+                    .wrapContentSize()
+                    .padding(16.dp)
+            ) {
+                Column(modifier = Modifier.padding(16.dp)) {
+                    Text(
+                        text = stringResource(R.string.nc_test_results),
+                        style = MaterialTheme.typography
+                            .titleMedium
+                    )
+                    Box(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .weight(1f, fill = false)
+                            .verticalScroll(rememberScrollState())
+                    ) {
+                        Column(modifier = Modifier.padding(top = 12.dp)) {
+                            Text(
+                                text = stringResource(R.string.nc_push_notification_message),
```

Comment:
Do not show the "Push notification is sent successfully..." string when the response was 404